src/esp/assets/ResourceManager.cpp

```diff
@@ -2184,5 +2184,147 @@ std::unique_ptr<MeshData> ResourceManager::createJoinedCollisionMesh(
+    }
+    Cr::Containers::Optional<Mn::Trade::MeshData> CHMesh = Mn::Trade::MeshData{
+        Mn::MeshPrimitive::Triangles,
+        {},
+        std::move(indices),
+        Mn::Trade::MeshIndexData{indices},
+        {},
+        std::move(positions),
+        {Mn::Trade::MeshAttributeData{Mn::Trade::MeshAttribute::Position,
+                                      Cr::Containers::arrayView(positions)}}};
+    Cr::Utility::Debug() << "ahh";
+    // Create a GenericMeshData
+    genCHMeshData = std::make_unique<GenericMeshData>(false);
+    genCHMeshData->setMeshData(*std::move(CHMesh));
+    genCHMeshData->BB = computeMeshBB(genCHMeshData.get());
+    genCHMeshData->uploadBuffersToGPU(false);
+
+    // Create CollisionMeshData and add to collisionMeshGroup vector
+    CollisionMeshData CHCollisionMesh = genCHMeshData->getCollisionMeshData();
+    collisionMeshGroup.push_back(CHCollisionMesh);
+
+    // register GenericMeshData in meshes_ dict
+    meshes_.emplace(meshes_.size(), std::move(genCHMeshData));
+    Cr::Utility::Debug() << "loop";
+  }
+  // make MeshMetaData
+  int meshEnd = meshes_.size() - 1;
+  MeshMetaData meshMetaData{meshStart, meshEnd};
+
+  // get original componentID (REVISIT)
+  int componentID = getMeshMetaData(filename).root.componentID;
+
+  // populate MeshMetaData root children
+  for (unsigned int p = 0; p < nConvexHulls; ++p) {
+    MeshTransformNode transformNode;
+    transformNode.meshIDLocal = p;
+    transformNode.componentID = componentID;
+    transformNode.materialIDLocal = 0;
+    meshMetaData.root.children.push_back(transformNode);
+  }
+
+  Cr::Utility::Debug() << "transforms MADE";
+
+  // default material for now
+  auto phongMaterial = gfx::PhongMaterialData::create_unique();
+  phongMaterial->specularColor = {1.0, 1.0, 1.0, 1.0};
+  phongMaterial->ambientColor = Magnum::Color4{1.0};
+  ;
+  phongMaterial->diffuseColor = Magnum::Color4{0.9};
+  ;
+
+  meshMetaData.setMaterialIndices(nextMaterialID_, nextMaterialID_);
+  shaderManager_.set(std::to_string(nextMaterialID_++),
+                     static_cast<gfx::MaterialData*>(phongMaterial.release()));
+
+  // make assetInfo
+  AssetInfo info{AssetType::PRIMITIVE};
+  info.requiresLighting = false;
+
+  // store the rotation to world frame upon load - currently superfluous
+  const quatf transform = info.frame.rotationFrameToWorld();
+  Magnum::Matrix4 R = Magnum::Matrix4::from(
+      Magnum::Quaternion(transform).toMatrix(), Magnum::Vector3());
+  meshMetaData.root.transformFromLocalToParent =
+      R * meshMetaData.root.transformFromLocalToParent;
+
+  // make LoadedAssetData corresponding to this asset
+  LoadedAssetData loadedAssetData{info, meshMetaData};
+  // TODO : need to free render assets associated with this object if collision
+  // occurs, otherwise leak! (Currently unsupported).
+  // if (resourceDict_.count(trajVisName) != 0) {
+  //   resourceDict_.erase(trajVisName);
+  // }
+
+  Cr::Utility::Debug() << "END";
```

Comment:
@ScottCSteinhauser printf debugging hint -- you [can use "exclamated" `!Debug{};` to add markers containing the file and line number](https://doc.magnum.graphics/corrade/classCorrade_1_1Utility_1_1Debug.html#Utility-Debug-source-location), without having to type arbitrary text ;)

Also, `Cr::Utility::Debug{}` is aliased to `Mn::Debug{}` for less typing.